### PR TITLE
Fix sign in page/set password page styles, refactoring

### PR DESCRIPTION
**Pull request by jasperhuangg**
There were too many unrelated changes happening in this PR so @marcaaron and I decided to break this up into a bunch of smaller PRs. I'll update the links individually once each PR is created. For now, this PR won't be merged but will simply be broken up into these PRs:
1. PR for the flex styles https://github.com/Expensify/Expensify.cash/pull/2192 [MERGED]
2. PR for refactoring the welcome text into its own component https://github.com/Expensify/Expensify.cash/pull/2190 [MERGED]
3. PR for fixing link fonts https://github.com/Expensify/Expensify.cash/pull/2237 [MERGED]
4. PR for refactoring `SigninPageLayout` to be used with single component forms https://github.com/Expensify/Expensify.cash/pull/2196 [MERGED]
5. PR for refactoring `LoginForm` to be a single component https://github.com/Expensify/Expensify.cash/pull/2363 [MERGED]
6. PR for fixing styles for the `SetPasswordPage` (will be renamed to `SetPasswordForm`) https://github.com/Expensify/Expensify.cash/pull/2211
7. PR for fixing CSS styles to allow native views to rely on width (as opposed to always defaulting to narrow).

### CODE REVIEW

**Comment by shawnborton** [comment]
Thanks @jasperhuangg! While we are here, I think it would be nice if the set password page looked more like the sign in page - same width, same centering, etc:
image

Desktop feels unnecessarily wide too, so perhaps we can fix that up as well?

**Comment by jasperhuangg** [comment]
Hey @shawnborton!

I rewrote the set password page to use an identical layout as the sign in page. I think it might look a bit too close, let me know what you think and if I need to get rid of anything. (see screenshots in the OP)

**Comment by shawnborton** [comment]
Nice, looks good to me design-wise. I like that we're just reusing the login/sign in styles.

**Review by shawnborton — DISMISSED** [review]

File: src/pages/setpassword/SetPasswordForm/SetPasswordFormNarrow.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordForm/SetPasswordFormNarrow.js:43]: Can we apply the DRY principle here somehow? These two components both use the same `submitForm()` method.

File: src/pages/setpassword/SetPasswordForm/SetPasswordFormWide.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordForm/SetPasswordFormWide.js:14]: What is different about this form compared to the "narrow" version? Splitting them out into two components makes it difficult to see.

File: src/pages/setpassword/SetPasswordForm/index.native.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordForm/index.native.js:3]: What if we have a large tablet like iPad Pro? Should we still show the "narrow" view?

**Review by marcaaron — CHANGES_REQUESTED** [review]
Looks good, but I think we can maybe apply the DRY principle here. If possible, I'd love to see these changes all happening in one file (`SetPasswordPage`) without all of these new files.

File: src/pages/setpassword/SetPasswordForm/index.native.js
Comment (shawnborton) [reply src/pages/setpassword/SetPasswordForm/index.native.js:3]: I think it would be great to make the form just one component, even if it means we have to make some design adjustments. For example, I'd be fine with the mobile view having everything left-aligned like the non-mobile view if that means we're using the same component.

File: src/pages/setpassword/SetPasswordForm/SetPasswordFormWide.js
Comment (jasperhuangg) [reply src/pages/setpassword/SetPasswordForm/SetPasswordFormWide.js:14]: I was just following the same pattern that the signin page is using, I can refactor it into a single component if you think that's best!

**Comment by jasperhuangg** [comment]
Hey @marcaaron!

I DRYed up the code and I think it should be good for another review. The design didn't change from before (I think my previous commits dismissed Shawn's review), so feel free to merge if you think everything looks good!

File: src/pages/setpassword/SetPasswordPageProps.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPageProps.js:3]: This file should be named using `camelCase`, use the `somethingPropTypes` convention, and be imported with the same name like this

```js
// setPasswordPagePropTypes.js
import setPasswordPagePropTypes from '../blah';
```

File: src/pages/setpassword/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPage.js:29]: Does this need to be it's own component?

File: src/pages/setpassword/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPage.js:50]: stateless function components should have a `displayName` added

```js
SetPasswordPage.displayName = 'SetPasswordPage';
```

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:55]: NAB, I appreciate the attempt to organize this by moving these files around. But it makes the code changes harder to review since we need to look at one file that has been removed completely and then another file that has been added - when actually there is a lot of overlap between them.

Ideally you can make changes like this so that they show up as a file rename instead of a deleted/created file pair that way we get a nice clean diff and can focus on what has actually changed instead of working hard to figure that out.

**Review by marcaaron — CHANGES_REQUESTED** [review]

**Comment by marcaaron** [comment]
>feel free to merge if you think everything looks good

Hey @jasperhuangg there is a HOLD on this PR. If you remove it then it can be merged with an approval.

File: src/pages/setpassword/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPage.js:16]: This is the 3rd use of this UI element so I think maybe it's time to create a reusable component out of this...? 🤔 

https://github.com/Expensify/Expensify.cash/blob/9b282f10de193d9a4115bc30f51e0dfdb4133050/src/pages/signin/LoginForm/LoginFormNarrow.js#L107-L109

https://github.com/Expensify/Expensify.cash/blob/9b282f10de193d9a4115bc30f51e0dfdb4133050/src/pages/signin/LoginForm/LoginFormWide.js#L98-L100

File: src/pages/setpassword/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPage.js:13]: By declaring this inside the component we will re-create it on each render. It would be better to create a new component and then use it where we are using `welcomeText`.

File: src/pages/SetPasswordPage.js
Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:55]: For sure, I've bookmarked that SO you sent me and will make sure I do this in the future!

File: src/pages/setpassword/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPage.js:30]: NAB, these props could _probably_ be passed explicitly there aren't an unreasonable number of them

File: src/pages/setpassword/SetPasswordPageProps.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPageProps.js:23]: I see all the above props being passed as `propTypes` to `SetPasswordPage`, but that's not the component that is receiving these props.

File: src/pages/setpassword/SetPasswordPageProps.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPageProps.js:29]: This looks like the only one that would also be passed to `SetPasswordForm` so I'm not sure it's worth splitting out these propTypes into it's own file after all.

File: src/pages/setpassword/SetPasswordPageProps.js
Comment (marcaaron) [reply src/pages/setpassword/SetPasswordPageProps.js:23]: In fact, it looks like nothing is connected to Onyx so none of these props will be available.

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:140]: We use a `compose()` method to combine` withOnyx()` and `withWindowDimensions()`

File: src/components/WelcomeText.js
Comment (marcaaron) [inline src/components/WelcomeText.js:24]: needs a `displayName`

File: src/components/WelcomeText.js
Comment (marcaaron) [inline src/components/WelcomeText.js:8]: we also export `windowDimensionsPropTypes` to use here

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:1]: This change looks unnecessary

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:3]: NAB, the change to put this on one line doesn't seem necessary

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:40]: `...windowDimensionsPropTypes`

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:56]: This was fine the way it was.

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:58]: This doesn't need to be removed.

**Review by marcaaron — CHANGES_REQUESTED** [review]
Looks good just have a few more clean up suggestions.

File: src/pages/SetPasswordPage.js
Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:56]: Ahh my bad, all these seemingly unnecessary changes were from running the eslint check with `--fix`. Will get rid of them!

Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:56]: I should just be able to revert that commit

Comment (marcaaron) [reply src/pages/SetPasswordPage.js:56]: Cool cool thanks! Auto-fixers can sure make a mess.

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:129]: Not sure if this is the effect you are going for, but I noticed that if we are viewing this on an iPad this "welcome screenshot" is no longer present.

We are using the "narrow" view on all iOS devices so this ends up looking like this...

**iPad**
Screen Shot 2021-03-26 at 8 36 24 AM

**iPhone**
Screen Shot 2021-03-26 at 8 38 38 AM

**Review by marcaaron — COMMENTED** [review]
PR looks good, thanks for the changes. I found one inconsistency on iPad that we should maybe address here. Other than that, I'm unsure why there is a HOLD on this PR. It looks ready to merge after the iPad issue is fixed.

File: src/pages/SetPasswordPage.js
Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:129]: Ahhh that is strange and definitely not what I was going for. So `isSmallScreenWidth` is dependent on the user agent, and not the actual screen width in pixels.

Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:129]: @marcaaron I noticed that this issue appears when you try to run it natively on iOS; I was wondering how you were able to access the set password page on iOS in this branch? Normally for web I access it via the URL.

Comment (marcaaron) [reply src/pages/SetPasswordPage.js:129]: yes, `isSmallScreenWidth` is just based on the screen width on any platform

>this issue appears when you try to run it natively on iOS

I'm using deep links to open this screen directly like this

```
npx uri-scheme open expensify-cash://setpassword/123456 --ios
```

Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:129]: @marcaaron Hmm tried it on my end with this branch and I wasn't able to reproduce the bug:

Simulator Screen Shot - iPad Pro (9 7-inch) - 2021-03-29 at 10 28 07

Is the main issue you're concerned about the image not showing up for you? Or did you want to change it so that we use the "wide" view for iPad?

Comment (marcaaron) [reply src/pages/SetPasswordPage.js:129]: The screenshot you are showing is the web version of the app shown in Safari app. The issue I am describing occurs in the native version of the app. Maybe try merging the `main` branch back into this branch and try again.

Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:129]: @marcaaron 

Ah! I uploaded the wrong screenshot (I've got like 50 screenshots on my desktop, my bad!)

Screen Shot 2021-03-30 at 1 30 27 PM

Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:129]: @marcaaron 

So the issue is that we're defaulting to the narrow view on all iOS devices, but this iOS device in particular has a width greater than the cut off for mobile devices. 

Since we put everything into a single component, I have to display the welcome screenshot conditionally based on the screen width.

I tried modifying `index.native.js` to use the screenWidth conditionally to display the narrow/wide views, but the wide view doesn't display correctly on iOS.

Screen Shot 2021-03-31 at 12 22 11 PM


A solution is to refactor everything to use the same pattern as the log in page, or to figure out why the styles aren't showing up correctly for the wide view on iOS (which could potentially be a deep rabbit hole).

Comment (jasperhuangg) [reply src/pages/SetPasswordPage.js:129]: I've refactored everything back into multiple components seeing that the sign in page follows this same pattern, and it makes more sense to have two pages that look nearly identical to have the same structure. This is necessary for everything to display properly seeing that we default to the narrow view for ios.

Screen Shot 2021-03-31 at 1 11 01 PM

**Comment by shawnborton** [comment]
Jumping in late here, but we should be using screen width to determine the layout and not the operating system. Is that what we're doing here?

File: src/libs/Navigation/AppNavigator/PublicScreens.js
Comment (marcaaron) [reply src/libs/Navigation/AppNavigator/PublicScreens.js:4]: This is not following the correct naming conventions as the import name should match the file extension.

File: src/pages/SetPasswordPage.js
Comment (marcaaron) [inline src/pages/SetPasswordPage.js:1]: Again, this entire file has been removed. Please modify what you need to without renaming unnecessarily.

File: src/pages/setpassword/SetPasswordForm.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordForm.js:12]: How did we end up here again? 😬  
We already established this could be achieved in a single file, no?

File: src/pages/setpassword/SetPasswordPageNarrow.js
Comment (marcaaron) [inline src/pages/setpassword/SetPasswordPageNarrow.js:16]: Not sure I will block on this change, but I don't think the code we have now is the correct pattern to use. Rather than blindly follow examples that already exist we should try to understand _why_ they exist and improve them.

Comment (marcaaron) [reply src/pages/setpassword/SetPasswordPageNarrow.js:16]: Mainly the idea that we need a Narrow and a Wide version of things... there is no clear reason why this can't exist in a single file that is using the screen width to present different views. It shares many of the same elements.

**Review by marcaaron — CHANGES_REQUESTED** [review]
Sorry @jasperhuangg I feel like this PR has regressed back to where we started and we're now going in circles. Things were very close, then we took a few steps backwards. Not sure why.

**Comment by marcaaron** [comment]
>Jumping in late here, but we should be using screen width to determine the layout and not the operating system. Is that what we're doing here?

+1 I'm not following why we are using a "narrow" view on native platforms, but maybe there is a clear reason for it? To me, the existing design of the `LoginForm` makes things complicated since we are presenting only the "narrow" view on native platforms which forces us to decentralize various things. If everyone agrees, I think we should ditch this pattern and not continue to follow it for the next screen in the sequence (set password).

We can also create a follow up to address that concern, but let's come to a decision before proceeding further.

**Pull request closed without merging**
